frontend/src/Components/ContentComponent.tsx

```diff
@@ -127,8 +128,39 @@ function updateMailbox(mailbox: HTMLSpanElement, setMailboxKey: (key: MailboxKey
   })
 }
 
-function updateMail(mail: HTMLSpanElement, setMailboxKey: (key: MailKey | null) => void, currentUsername?: string) {
-  // check processed
+function renderWithTheme(container: HTMLElement, content: React.ReactNode, appState: AppState) {
+  const root = createRoot(container)
+  elementToRoot.set(container, root)
+
+  const renderContent = () => {
+    if (!elementToRoot.has(container)) return
+    root.render(<FakeRoot appState={appState}>{content}</FakeRoot>)
+  }
+
+  renderContent()
+
+  const observer = new MutationObserver(renderContent)
```

Comment:
maybe making theme an observable property in the `AppState` would simplify things, including watching for the changes here?